File: row-change-event-channel/src/test/java/nl/haploid/octowight/TestData.java
package nl.haploid.octowight;

import java.util.Random;
import java.util.UUID;

public class TestData {

	public static final String ATOM_LOCUS = "sunnydale";

	public static final String ATOM_TYPE = "spike";

	public static AtomChangeEvent atomChangeEvent() {
		final AtomChangeEvent event = new AtomChangeEvent();
		event.setAtomId(nextLong());
		event.setAtomLocus(ATOM_LOCUS);
		event.setAtomType(ATOM_TYPE);
		return event;
	}

	public static long nextLong() {
		return new Random().nextLong();
	}

	public static String message() {
		return UUID.randomUUID().toString();
	}

	public static String topic() {
		return UUID.randomUUID().toString();
	}
}

File: event-emitter/src/test/scala/nl/haploid/octowight/TestData.scala
package nl.haploid.octowight

import java.util.{Random, UUID}

object TestData {
  val ATOM_LOCUS = nextString
  val ATOM_TYPE = nextString

  def atomChangeEvent = {
    val event = new AtomChangeEvent
    event.setAtomId(nextLong)
    event.setAtomOrigin(ATOM_LOCUS)
    event.setAtomCategory(ATOM_TYPE)
    event
  }

  def nextLong: Long = new Random().nextLong

  def nextString = UUID.randomUUID.toString

  def message = nextString

  def topic = nextString
}
